internal/connector/internal/services/connectors.go

```diff
@@ -351,16 +352,21 @@ func (k *connectorsService) Update(ctx context.Context, resource *dbapi.Connecto
 	return nil
 }
 
-func (k *connectorsService) SaveStatus(ctx context.Context, resource dbapi.ConnectorStatus) *errors.ServiceError {
+func (k *connectorsService) SaveStatus(ctx context.Context, connectorStatus dbapi.ConnectorStatus) *errors.ServiceError {
 	dbConn := k.connectionFactory.New()
-	if err := dbConn.Model(resource).Save(resource).Error; err != nil {
+	if err := dbConn.Where("deleted_at IS NULL").Save(&connectorStatus).Error; err != nil {
```

Comment:
Why add `Where("deleted_at IS NULL")`? Unless the query is 'unscoped' or 'raw', gorm should automatically add it.

Reply:
Yeah you are right probably it was a leftover before discovering that using `Model()` with `Save()` is not supported and can lead to undefined behavior (as per documentation first NOTE: https://gorm.io/docs/update.html)

Reply:
Well actually it seems that `deleted_at` is not take into account when `Save()` is used at least in the gorm version we use :shrug: